### DISCUSSION TITLE
[MRTK3] How to trigger gaze active hover events without hand raise

### DISCUSSION THREAD

**Discussion by pajama** [microsoft/MixedRealityToolkit-Unity#11108] [General]
Currently GazeActiveHover is only triggered if a hand is tracked. How do I trigger these state events on buttons without tracking the hand.

HL2

## Unity editor version
2020.3.38f1

## Mixed Reality Toolkit release version
MRTK3.pre11

Thank you!

**Comment by Zee2** [comment]
You can use IsGazeHovered.

**Comment by pajama** [comment]
@Zee2 Thanks for responding! 

In the statevisualizer I hooked up the same animation to PassiveHover as the one in ActiveHover, but it still doesn't trigger any animation with just gaze.

I did validate that IsGazeHovered is firing on the PressableButton, I'm just not sure how to get that event to trigger the hover animation on the statevisualizer. I want to use the same playable api. Is that possible, or do I need a completely separate animation setup to do this?

**Comment by Zee2** [comment]
The animation should work in PassiveHover on StateVisualizer; that's actually how we already drive the subtle content "bump" that appears when you gaze at a button without your hand up. If that didn't work, you wouldn't see any of the buttons animate when you gazed at them. Curious what is causing that not to work for you... have you tried removing your animation from ActiveHover and putting it in PassiveHover exclusively?

**Reply by pajama** [reply]
@Zee2 woah removing the effect from ActiveHover allows it to trigger in PassiveHover! I don't believe PassiveHover is working at all for me in the default configuration.

**Reply by Zee2** [reply]
What do you mean "default configuration"?

On the side; there might be a bug when trying to apply the same animation twice... when ActiveHover is active, PassiveHover is also active, so any effects in both will be applied twice. This shouldn't be an issue (in fact, it's a feature to be able to additively combine simultaneous effects) but maybe there's a bug somewhere in there.

**Reply by pajama** [reply]
Default meaning PressableButton_32x32mm_Circular_IconOnly prefab
it turns out the passive animation does play, it was just too subtle for me to know with how far we have the button. 

I did verify however that it does not play if both the passive and active hover animations are the same.
